### PR TITLE
Auto healing Foreman Proxy Content blogpost

### PR DESCRIPTION
**Pull request by sean797**

### CODE REVIEW

File: _posts/2017-08-23-auto-healing-foreman-proxy.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
+---
+layout: post
+title: Creating a auto healing Foreman Proxy
```
Comment (evgeni) [inline _posts/2017-08-23-auto-healing-foreman-proxy.md]: an*

File: _posts/2017-08-23-auto-healing-foreman-proxy.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+```json
+<SNIP>
+
+"provisioners": [
+    {
+        "type": "shell",
+        "inline": [
+            "sed -i 's/console=ttyS0,115200n8 console=tty0/console=tty0 console=ttyS0,115200n8/' /etc/default/grub",
+            "sudo grub2-mkconfig -o /boot/grub2/grub.cfg",do subscription-manager clean",
+            "rpm -q katello-ca-consumer-katello.example.com 2>/dev/null || sudo rpm -Uvh http://katello.example.com/pub/katello-ca-consumer-latest.noarch.rpm",
```
Comment (evgeni) [inline _posts/2017-08-23-auto-healing-foreman-proxy.md]: Won't that break if you ever decide to update the certs and your AMI has the older already installed?

File: _posts/2017-08-23-auto-healing-foreman-proxy.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+1. An Auto Scaling Group per Smart Proxy attached to the ELB, limited to 1 instance
+1. A route53 DNS record for the ELB with a CNAME to the instance
+
+## Conclusion
+
+There are a couple of gotchas with this approach, but none that affected us. 
+* Every time the instance is rebuilt it has to re-sync Content from the Katello server, this could become an issue depending on scale, you could look into storing `/var/lib/pulp/` & `/var/lib/mongodb/` on a persistent volume to get around this, but we didn't.
+* We were only using the Smart Proxy Pulp & OpenSCAP features, other features like DHCP, DNS, TFTP would require persistent volumes to ensure they don't get blown away when the instance is rebuilt.
+
+We have now a repeatable Foreman Smart Proxy build defined in git, enabling us to tare down from one cloud provider and rebuild it on another if required, but for us the most valuable thing is integrating Foreman Smart Proxies to our build process for all our other applications and servers.
```
Comment (evgeni) [inline _posts/2017-08-23-auto-healing-foreman-proxy.md]: s/tare/tear/

File: _posts/2017-08-23-auto-healing-foreman-proxy.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+1. A security group allowing ports required for the Smart Proxy Content.
+1. Upload certs.tar to S3 for our capsules.
+1. An IAM profile to allow the instance to pull down the certs.tar from S3
+1. An Elastic Load Balancer with listeners for all the ports Smart Proxy uses, this ensures Katello & Clients can use the same hostname to communicate with it.
+1. Instance using the above "foreman-proxy" AMI with user data that:
+   * Sets the hostname to that of the ELB
+   * Pulls down the certs.tar from S3
+   * Runs foreman-installer via an [Ansible role](https://galaxy.ansible.com/sean797/foreman_installer)
+   * Hits https://katello.example.com/katello/api/capsules/:id/content/sync to kick off a Content Sync
+   * Reverts hostname change back
```
Comment (evgeni) [inline _posts/2017-08-23-auto-healing-foreman-proxy.md]: Why is that required?

File: _posts/2017-08-23-auto-healing-foreman-proxy.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+1. An Auto Scaling Group per Smart Proxy attached to the ELB, limited to 1 instance
+1. A route53 DNS record for the ELB with a CNAME to the instance
+
+## Conclusion
+
+There are a couple of gotchas with this approach, but none that affected us. 
+* Every time the instance is rebuilt it has to re-sync Content from the Katello server, this could become an issue depending on scale, you could look into storing `/var/lib/pulp/` & `/var/lib/mongodb/` on a persistent volume to get around this, but we didn't.
+* We were only using the Smart Proxy Pulp & OpenSCAP features, other features like DHCP, DNS, TFTP would require persistent volumes to ensure they don't get blown away when the instance is rebuilt.
```
Comment (evgeni) [inline _posts/2017-08-23-auto-healing-foreman-proxy.md]: Also no Puppet(Master/CA), right?

File: _posts/2017-08-23-auto-healing-foreman-proxy.md (hunk 4 above)
Comment (sean797) [reply _posts/2017-08-23-auto-healing-foreman-proxy.md]: It isn't related to Katello but other reasons like logging software also running on the same machine. Maybe I should remove it, as its not really relevant?

File: _posts/2017-08-23-auto-healing-foreman-proxy.md (hunk 2 above)
Comment (sean797) [reply _posts/2017-08-23-auto-healing-foreman-proxy.md]: Yes, but we rebuild the AMI every 21 days anyway and could easier rebuild on-demand.

Comment (evgeni) [reply _posts/2017-08-23-auto-healing-foreman-proxy.md]: I was more pointing at "why not use `yum install http://..` unconditionally"

File: _posts/2017-08-23-auto-healing-foreman-proxy.md (hunk 4 above)
Comment (evgeni) [reply _posts/2017-08-23-auto-healing-foreman-proxy.md]: Maybe "Reverts hostname change back, to meet expectations of the internal tooling, CMDB, etc"?

**Comment by sean797** [comment]
Thanks @evgeni 👍 updated

**Review by evgeni — APPROVED** [review]

File: _posts/2017-08-23-auto-healing-foreman-proxy.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
+---
+layout: post
+title: Creating an auto healing Foreman Proxy
+author: Sean O'Keeffe
```
Comment (GregSutcliffe) [inline _posts/2017-08-23-auto-healing-foreman-proxy.md]: Minor point, but a date field is needed for the RSS feed, I think.

**Comment by sean797** [comment]
Thanks @GregSutcliffe. Updated, also renamed the file to today's date :)

**Comment by GregSutcliffe** [comment]
Ugh, sorry for the delay - thx @sean797!
